modules/fx/arrow-fx/src/main/kotlin/arrow/fx/typeclasses/Concurrent.kt

```diff
@@ -774,9 +780,113 @@ interface Concurrent<F> : Async<F> {
     }
 
   /**
-   * Creates a variable [MVar] to be used for thread-sharing, initialized to a value [a]
+   * Create a pure [Promise] that is empty and can be filled exactly once.
+   *
+   * ```kotlin:ank:playground
+   * import arrow.Kind
+   * import arrow.fx.*
+   * import arrow.fx.extensions.io.concurrent.concurrent
+   * import arrow.fx.typeclasses.Concurrent
+   * import arrow.fx.typeclasses.seconds
+   *
+   * fun main(args: Array<String>) {
+   *   fun <F> Concurrent<F>.promiseExample(): Kind<F, Unit> =
+   *     //sampleStart
+   *     fx.concurrent {
+   *       val promise = !Promise<String>()
+   *       val (join, cancel) = !sleep(1.seconds)
+   *         .followedBy(promise.complete("Hello World!"))
+   *         .fork()
+   *       val message = !join
+   *       message
+   *     }
+   *
+   *   //sampleEnd
+   *   IO.concurrent().promiseExample()
+   *     .fix().unsafeRunSync()
+   * }
+   * ```
+   *
+   * @see Promise for more details on usage
+   */
+  fun <A> Promise(): Kind<F, Promise<F, A>> =
```

Comment:
Should these be part of the interface or extfuns? Not much difference because we control them, still.

Reply:
This is temporary sugar that mimics KEEP-87 support in the DSL.